pkgdb/src/search/params.cc

```diff
@@ -134,13 +134,26 @@ pkgdb::PkgQueryArgs &
 SearchQuery::fillPkgQueryArgs( pkgdb::PkgQueryArgs & pqa ) const
 {
   /* XXX: DOES NOT CLEAR FIRST! We are called after global preferences. */
-  pqa.name             = this->name;
-  pqa.pname            = this->pname;
-  pqa.version          = this->version;
-  pqa.semver           = this->semver;
-  pqa.partialMatch     = this->partialMatch;
-  pqa.partialNameMatch = this->partialNameMatch;
-  pqa.limit            = this->limit;
+  pqa.name    = this->name;
+  pqa.pname   = this->pname;
+  pqa.version = this->version;
+  pqa.semver  = this->semver;
+  // These will both write to `path`, so only one of them can be present,
+  // which is the same invariant that we already put in place on the `flox`
+  // side.
+  if ( this->partialMatch.has_value() )
+    {
+      AttrPath path = splitAttrPath( *this->partialMatch );
+      if ( path.size() > 1 ) { pqa.relPath = path; }
+      else { pqa.partialMatch = this->partialMatch; }
```

Comment:
Is there a chance where we have a partial match **and** a `path.size > 1` or should `partialMatch` always be set?

Same question wrt `partialNameMatch`

Reply:
There's no `path` query parameter, so we're always going to get relative paths sent via these parameters (if a relative path is sent at all). Anything that has more than one component is considered to be a relative path instead of some kind of match on the name. So, in short, we're explicitly saying "no, you can't have a `partial(Name)Match` on anything with more than one component."

Reply:
Got it maybe a comment for the uninitiated?